src/cls/rbd/cls_rbd.cc

```diff
@@ -2499,9 +2499,10 @@ int object_map_resize(cls_method_context_t hctx, bufferlist *in, bufferlist *out
   }
 
   size_t orig_object_map_size = object_map.size();
+  auto it = object_map.begin();
   if (object_count < orig_object_map_size) {
     for (uint64_t i = object_count + 1; i < orig_object_map_size; ++i) {
-      if (object_map[i] != default_state) {
+      if (*(it + i) != default_state) {
```

Comment:
Nit: I doubt this actually saves any time and it less readable. Why not set ```it``` to ```object_count + 1``` offset and then increment it in the loop.

Reply:
done.  thanks